apps/examples/iotivity_simpleserver/simpleServer.c

```diff
@@ -0,0 +1,1012 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+//******************************************************************
+//
+// Copyright 2014 Intel Mobile Communications GmbH All Rights Reserved.
+//
+//-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+
+/**
+   * @testcase		   :iotivity_lwip_tc
+   * @brief		   :Discovery resources and devices using IoTivity
+   * @scenario		   :
+   * @apicovered	   :
+   * @precondition	   :At Linux PC,	1. Build an IoTivity client
+   *			   :			 $ ./occlient -t <tc number>
+   *			   :At test board,	1. Make a binary with 'iotivity_simpleserver' example configuration and flash
+   *			   :				($ ./configure.sh artik053/iotivity)
+   *			   :			2. Run the TASH
+   *			   :			3. Connect to Wi-Fi and get an IP address using DHCP
+   *			   :			 TASH>> simpleServer
+   * @postcondition	   :
+   */
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <signal.h>
+#include <pthread.h>
+
+#include "ocstack.h"
+#include "logger.h"
+#include "platform_features.h"
+#include "ocpayload.h"
+#include "ocserver.h"
+
+/* For WiFi -- */
+#include <net/if.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <apps/netutils/netlib.h>
+
+/****************************************************************************
+ * Definitions
+ ****************************************************************************/
+
+#define IFHWADDRLEN 6
+
+#if defined(LWIP_HAVE_LOOPIF) && LWIP_HAVE_LOOPIF == 1
```

Comment:
When LWIP_HAVE_LOOPIF is not defined, this is defined with 0 in os/include/net/lwip/opt.h. So, this is always defined but can have different value. Let's remove checking definition.
#if LWIP_HAVE_LOOP == 1